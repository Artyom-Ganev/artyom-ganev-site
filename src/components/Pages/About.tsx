import React, { FunctionComponent } from 'react';

import { bio } from 'src/assets';
import { PageContainer } from 'src/container';

import styles from './About.module.scss';

/**
 * About page component
 */
const About: FunctionComponent = () => (
  <PageContainer title='Artyom Ganev'>
    {bio.data.map((value: string, key: number) => (
      <div key={key} className={styles.item}>
        {value}
      </div>
    ))}
  </PageContainer>
);

export default About;
