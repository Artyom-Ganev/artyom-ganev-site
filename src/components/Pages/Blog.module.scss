.itemContainer {
  padding-bottom: 0.5em;
  padding-left: 0.8em;
  letter-spacing: 1px;
  word-spacing: 1px;
}

.title {
  font-weight: bold;
  font-style: italic;
  margin: 0.5em 0;
}

.item {
  color: #343648;
}
