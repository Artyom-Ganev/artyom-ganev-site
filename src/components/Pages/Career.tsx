import React, { ReactNode } from 'react';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import * as actions from 'src/actions';
import { BLANK, REL } from 'src/const';
import { AbstractDataContainer } from 'src/container';
import { ICareer, ICareerItems } from 'src/interface/ICareer';
import { IDataProps } from 'src/interface/IData';
import { ISource } from 'src/interface/ISource';
import IState from 'src/interface/IState';
import { createSource } from 'src/model';
import { DateUtil } from 'src/utils';
import styles from './Career.module.scss';

interface IProps<TData> extends IDataProps<TData> {
  appLoading: (payload: { loading: boolean }) => void;
  careerLoadList: (payload: { items: TData[] }) => void;
}

const itemsSelector = createSelector(
  ({ career }: IState) => career,
  (career: { items: ICareerItems }) => Object.values(career.items)
);

const mapStateToProps = (state: IState): IDataProps<ICareer> => ({ items: itemsSelector(state) });

const actionCreators = {
  appLoading: actions.appLoading,
  careerLoadList: actions.careerLoadList,
};

/**
 * Career page
 */
class Career extends AbstractDataContainer<ICareer, IProps<ICareer>> {
  private readonly source: ISource;

  constructor(props: IProps<ICareer>) {
    super(props);
    const { appLoading, careerLoadList } = props;
    this.source = createSource<ICareer>()
      .endpoint('career')
      .beforeLoad(() => {
        appLoading({ loading: true });
      })
      .afterLoad((data: ICareer[]) => {
        careerLoadList({ items: data });
      })
      .build();
  }

  protected getSource = (): ISource => this.source;

  private prepareTitle = (site: string, title: string): ReactNode => {
    const header = <h3>{title}</h3>;
    return site ? (
      <a href={site} target={BLANK} rel={REL} title='Click for details'>
        {header}
      </a>
    ) : (
      <h3>{title}</h3>
    );
  };

  /**
   * Career items markup
   */
  protected getContent = (careerList: ICareer[]): ReactNode => {
    return careerList.map(({ id, site, title, startDate, endDate, post, description, tools }: ICareer) => (
      <div key={id} className={styles.item}>
        {this.prepareTitle(site, title)}
        <div className={styles.dates}>{DateUtil.prepareDates(startDate, endDate)}</div>
        <div>Post:&nbsp;{post}</div>
        <div>{description}</div>
        <div className='flexBox flexColumn'>
          <div>Tools:&nbsp;{tools}</div>
        </div>
      </div>
    ));
  };
}

export default connect(mapStateToProps, actionCreators)(Career);
