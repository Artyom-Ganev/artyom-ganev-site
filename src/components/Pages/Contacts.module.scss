.container {
  line-height: 24px;
  letter-spacing: 1px;
  word-spacing: 1px;
  padding-left: 0.8em;
}

.item {
  text-align: right;
}
