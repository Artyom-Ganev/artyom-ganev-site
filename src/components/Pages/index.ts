export { default as Main } from './Main';
export { default as About } from './About';
export { default as Blog } from './Blog';
export { default as Career } from './Career';
export { default as Contacts } from './Contacts';
export { default as NotFound } from './NotFound';
