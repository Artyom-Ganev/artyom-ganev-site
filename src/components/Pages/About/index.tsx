import Typography from '@material-ui/core/Typography';
import React, { FunctionComponent } from 'react';
import { bio } from 'src/assets';
import { PageContainer } from 'src/components/Pages';
import useStyles from './styles';

/**
 * About page component
 */
const About: FunctionComponent = () => {
  const classes = useStyles();
  return (
    <PageContainer title='Artyom Ganev'>
      {bio.data.map((value: string, key: number) => (
        <Typography key={key} className={classes.aboutPageBlock} paragraph>
          {value}
        </Typography>
      ))}
    </PageContainer>
  );
};

export default About;
