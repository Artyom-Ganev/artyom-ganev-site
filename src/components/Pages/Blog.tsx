import Box from '@material-ui/core/Box';
import Card from '@material-ui/core/Card';
import Typography from '@material-ui/core/Typography';
import { PageContainer, useBlogItems } from 'components/Pages';
import useStyles from 'components/Pages/Blog.styles';
import { BLANK, BLOG, REL } from 'const';
import { BlogModel } from 'model';
import React, { FC } from 'react';

const Blog: FC = () => {
  const classes = useStyles();
  const items: BlogModel[] = useBlogItems();
  return (
    <PageContainer title={BLOG.name} Icon={BLOG.Icon}>
      <Box display='flex' flexDirection='column'>
        {items.map((item: BlogModel) => {
          const { id, title, link, linkCaption } = item;
          // TODO: move to separate component
          return (
            <Box key={id} mr={2} mb={2}>
              <Card raised>
                <Box mr={2} mb={2} px={2} pt={2} flexGrow={1}>
                  <Typography variant='h5' paragraph className={classes.blogListDate}>
                    {item.formatDate()}
                  </Typography>
                  <Typography variant='h6' paragraph>
                    {title}
                  </Typography>
                  {link && (
                    <Box display='flex'>
                      <a href={link} target={BLANK} rel={REL} title='Click for details'>
                        <Typography variant='h6' paragraph>
                          {linkCaption}
                        </Typography>
                      </a>
                    </Box>
                  )}
                </Box>
              </Card>
            </Box>
          );
        })}
      </Box>
    </PageContainer>
  );
};

export default Blog;
