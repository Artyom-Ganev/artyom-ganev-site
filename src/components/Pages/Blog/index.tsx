import Box from '@material-ui/core/Box';
import Card from '@material-ui/core/Card';
import Typography from '@material-ui/core/Typography';
import React, { FunctionComponent } from 'react';
import { PageContainer } from 'src/components/Pages';
import { useBlogItems } from 'src/components/Pages/Blog/hooks';
import { BLANK, BLOG, REL } from 'src/const';
import { BlogModel } from 'src/model';
import useStyles from './styles';

const Blog: FunctionComponent = () => {
  const classes = useStyles();
  const items: BlogModel[] = useBlogItems();
  return (
    <PageContainer title={BLOG.name} Icon={BLOG.Icon}>
      <Box display='flex' flexWrap='wrap' justifyContent='space-between'>
        {items.map((item: BlogModel) => {
          const { id, title, link, linkCaption } = item;
          // TODO: move to separate component
          return (
            <Card key={id} className={classes.blogListItem} raised>
              <Typography variant='h5' paragraph className={classes.blogListDate}>
                {item.formatDate()}
              </Typography>
              <Typography variant='h6' paragraph>
                {title}
              </Typography>
              {link && (
                <Box display='flex'>
                  <a href={link} target={BLANK} rel={REL} title='Click for details'>
                    <Typography variant='h6' paragraph>
                      {linkCaption}
                    </Typography>
                  </a>
                </Box>
              )}
            </Card>
          );
        })}
      </Box>
    </PageContainer>
  );
};

export default Blog;
