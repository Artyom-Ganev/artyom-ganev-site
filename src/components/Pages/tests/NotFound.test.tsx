import { combineReducers, createStore } from 'redux';
import React from 'react';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { NotFound } from 'src/components/Pages';
import { IAppState } from 'src/interface/IState';

test('NotFound', () => {
  const app = (): IAppState => ({ loading: false });

  const store = createStore(combineReducers({ app }));

  const component = renderer.create(
    <Provider store={store}>
      <HashRouter>
        <NotFound />
      </HashRouter>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
