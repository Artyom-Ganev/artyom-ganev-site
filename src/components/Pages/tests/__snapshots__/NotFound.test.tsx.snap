// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound 1`] = `
<div
  className="flexBox flexColumn container wide"
>
  <h2>
    Page Not Found
  </h2>
  <div
    className="container"
  >
    <a
      href="#/"
      onClick={[Function]}
    >
      Go to main page
    </a>
  </div>
</div>
`;
